src/main/java/org/testng/TestNG.java

```diff
@@ -1368,11 +1376,11 @@ private SuiteRunner createSuiteRunner(XmlSuite xmlSuite) {
         m_testRunnerFactory,
         m_useDefaultListeners,
         m_methodInterceptors,
-        m_invokedMethodListeners,
-        m_testListeners,
-        m_classListeners);
+        Lists.newArrayList(m_invokedMethodListeners.values()),
```

Comment:
Sure. You can do it in 2 steps/commits: 1. fix it as minimal as possible, 2. Clean technical dept 👍